chromium/browser_type_test.go

```diff
@@ -32,101 +32,111 @@ import (
-		t.Parallel()
-
-		const headlessFlag = "headless"
-
-		var (
-			bt    BrowserType
-			lopts = &common.LaunchOptions{}
-		)
-
-		flags := bt.flags(lopts)
-		require.Contains(t, flags, headlessFlag)
-
-		_, ok := flags[headlessFlag].(bool)
-		require.Truef(t, ok, "%q should be a bool", headlessFlag)
-
-		lopts.Headless = false
-		assert.Falsef(t,
-			flags[headlessFlag].(bool),
-			"%q should also be false if launch options Headless is false", headlessFlag,
-		)
-
-		flags = bt.flags(&common.LaunchOptions{
-			Headless: true,
+	testCases := []struct {
+		flag                      string
+		changeOpts                *common.LaunchOptions
+		expInitVal, expChangedVal interface{}
+		pre                       func(t *testing.T)
+		post                      func(t *testing.T, flags map[string]interface{})
+	}{
+		{
+			flag:          "auto-open-devtools-for-tabs",
+			expInitVal:    false,
+			changeOpts:    &common.LaunchOptions{Devtools: true},
+			expChangedVal: true,
+		},
+		{
+			flag:          "browser-arg",
+			expInitVal:    nil,
+			changeOpts:    &common.LaunchOptions{Args: []string{"browser-arg=value"}},
+			expChangedVal: "value",
+		},
+		{
+			flag:          "browser-arg-flag",
+			expInitVal:    nil,
+			changeOpts:    &common.LaunchOptions{Args: []string{"browser-arg-flag"}},
+			expChangedVal: "",
+		},
+		{
+			flag:       "browser-arg-trim-double-quote",
+			expInitVal: nil,
+			changeOpts: &common.LaunchOptions{Args: []string{
+				`   browser-arg-trim-double-quote =  "value  "  `}},
+			expChangedVal: "value  ",
+		},
+		{
+			flag:       "browser-arg-trim-single-quote",
+			expInitVal: nil,
+			changeOpts: &common.LaunchOptions{Args: []string{
+				`   browser-arg-trim-single-quote=' value '`}},
+			expChangedVal: " value ",
+		},
+		{
+			flag:       "browser-args",
+			expInitVal: nil,
+			changeOpts: &common.LaunchOptions{Args: []string{
+				"browser-arg1='value1", "browser-arg2=''value2''", "browser-flag"}},
+			post: func(t *testing.T, flags map[string]interface{}) {
+				assert.Equal(t, "'value1", flags["browser-arg1"])
+				assert.Equal(t, "'value2'", flags["browser-arg2"])
+				assert.Equal(t, "", flags["browser-flag"])
+			},
+		},
+		{
+			flag:       "enable-use-zoom-for-dsf",
+			expInitVal: false,
+			pre: func(t *testing.T) {
+				if runtime.GOOS != "darwin" {
+					t.Skip()
+				}
+			},
+		},
+		{
+			flag:          "headless",
+			expInitVal:    false,
+			changeOpts:    &common.LaunchOptions{Headless: true},
+			expChangedVal: true,
+			post: func(t *testing.T, flags map[string]interface{}) {
+				extraFlags := []string{"hide-scrollbars", "mute-audio", "blink-settings"}
+				for _, f := range extraFlags {
+					assert.Contains(t, flags, f)
+				}
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.flag, func(t *testing.T) {
+			t.Parallel()
+			if tc.pre != nil {
+				tc.pre(t)
+			}
+
+			var (
+				bt    BrowserType
+				lopts = &common.LaunchOptions{}
```

Comment:
Maybe pass this directly to `bt.flags` instead of declaring a variable here?

Reply:
Resolved in 346dab6.